src/Nova/Invoice.php

```diff
@@ -0,0 +1,102 @@
+
+    public static $title = 'invoice_number';
+
+    public static $search = [
+        'invoice_number',
+    ];
+
+    public static function indexQuery(NovaRequest $request, $query)
+    {
+        if ($request->user()->hasRole([
+            'Admin',
+            'Owner',
+            'Accountant',
+            'Executive',
+            'Reservation Manager',
+            'Reservationist',
+        ])) {
+            return $query;
+        }
+
+        return $query->whereHas('order', function ($orderlocation) use ($request) {
+            return $orderlocation->whereIn('location_id', $request->user()->locations->pluck('id'));
+        });
+    }
+
+    public static $group = 'Operations';
+
+    public function fieldsForIndex(NovaRequest $request)
+    {
+        return array_filter([
+            ID::make()->sortable(),
+            Text::make('Invoice Number')->sortable(),
+            nova('order') ? BelongsTo::make('Order', 'order', nova('order'))->sortable() : null,
+            nova('customer') ? BelongsTo::make('Customer', 'customer', nova('customer'))->sortable() : null,
+            Currency::make('Amount')->asMinorUnits()->sortable(),
+            Date::make('Invoiced At', 'invoiced_at')->sortable(),
+        ]);
+    }
+
+    public function fields(Request $request)
+    {
+        return array_filter([
+            Text::make('Invoice Number')->exceptOnForms(),
+            nova('order') ? BelongsTo::make('Order', 'order', nova('order'))->hideWhenUpdating() : null,
+            nova('customer') ? BelongsTo::make('Customer', 'customer', nova('customer'))->exceptOnForms() : null,
+            Currency::make('Amount')->asMinorUnits()
+                ->step('0.01')
+                ->resolveUsing(function ($value) {
+                    return $value / 100;
+                })
+                ->fillUsing(function ($request, $model, $attribute) {
+                    $model->$attribute = $request->$attribute * 100;
+                }),
+            Text::make('Note'),
+            DateTime::make('Invoiced At', 'invoiced_at')->exceptOnForms(),
+            Date::make('Due Date', 'due_at'),
+
+            // Will need a 'Send Invoice' button
+
+            nova('payment') ? HasMany::make('Payments', 'payments', nova('payment')) : null,
+
+            new Panel('Data Fields', $this->dataFields()),
+        ]);
+    }
+
+    protected function dataFields(): array
+    {
+        return array_filter([
+            ID::make(),
+            nova('user') ? BelongsTo::make('Created By', 'creator', nova('user'))->exceptOnForms() : null,
+            DateTime::make('Created At')->exceptOnForms(),
+            nova('user') ? BelongsTo::make('Updated By', 'updater', nova('user'))->exceptOnForms() : null,
+            DateTime::make('Updated At')->exceptOnForms(),
+        ]);
+    }
+
+    public function filters(Request $request)
+    {
+        return [
+            new Filters\OrderLocation,
```

Comment:
@drewroberts Which package contains this filter? Thank you.

Reply:
I don't know yet. It will probably end up in the https://github.com/tipoff/locations package

Reply:
Yeah, let's go with that one